remove-webhook/adjust-downstream-webhook.sh

```diff
@@ -0,0 +1,136 @@
+
+Users also need to ensure they have the rancher-charts repo in the local Helm index.
+EOF
+}
+
+if [ "$1" == "-h" ]; then
+  usage
+  exit 0
+fi
+
+delete_webhook() {
+  cluster="$1"
+  current_chart=$(helm list -n cattle-system -l name=rancher-webhook | tail -1 | cut -f 6)
+  echo "Deleting $current_chart from cluster $cluster."
+  helm uninstall rancher-webhook -n cattle-system
+}
+
+replace_webhook() {
+  cluster="$1"
+  new_version="$2"
+
+  echo "Updating the agent to make it remember the min version $new_version of rancher-webhook, so that it can deploy it when needed in the future in cluster $cluster."
+  kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION="$new_version"
+
+  helm get values -n cattle-system rancher-webhook -o yaml > current_values.yaml
+  echo "Re-installing rancher-webhook to use $new_version in cluster $cluster."
+  helm upgrade --install rancher-webhook rancher-charts/rancher-webhook -n cattle-system --version "$new_version" --values current_values.yaml
+  rm -f current_values.yaml
+}
+
+adjust_webhook() {
+  cluster="$1"
+  rancher_version="$2"
+
+  if [[ "$rancher_version" =~ 2\.6\.13 ]]; then
+    replace_webhook "$cluster" 1.0.9+up0.2.10
+  elif [[ "$rancher_version" =~ 2\.6\.[0-9]$ ]] || [[ "$rancher_version" =~ 2\.6\.1[0-2]$ ]]; then
+    delete_webhook "$cluster"
+  elif [[ "$rancher_version" =~ 2\.7\.[0-1]$ ]]; then
+    delete_webhook "$cluster"
+  elif [[ "$rancher_version" =~ 2\.7\.2 ]]; then
+    replace_webhook "$cluster" 2.0.2+up0.3.2
+  elif [[ "$rancher_version" =~ 2\.7\.3 ]]; then
+    replace_webhook "$cluster" 2.0.3+up0.3.3
+  elif [[ "$rancher_version" =~ 2\.7\.4 ]]; then
+    replace_webhook "$cluster" 2.0.4+up0.3.4
+  elif [[ "$rancher_version" =~ 2\.[7-9]\..* ]]; then
+    # This matches anything else above 2.7, including 2.8.x and 2.9.x.
+    echo "No need to delete rancher-webhook, given Rancher version $rancher_version."
+    echo "Ensuring CATTLE_RANCHER_WEBHOOK_MIN_VERSION is set to an empty string."
+    kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION=''
+  else
+    echo "Nothing to do, given Rancher version $rancher_version."
+  fi
+}
+
+if [ -n "$DEBUG" ]
+then
+  set -x
+fi
+
+if [[ -z "$RANCHER_TOKEN" || -z "$RANCHER_URL" ]]
+then
+  echo "Required environment variables aren't properly set."
+  usage
+  exit 1
+fi
+
+tlsVerify="$1"
+
+kubeconfig="
+apiVersion: v1
+kind: Config
+clusters:
+- name: \"local\"
+  cluster:
+    server: \"$RANCHER_URL\"
+
+users:
+- name: \"local\"
+  user:
+    token: \"$RANCHER_TOKEN\"
+
+
+contexts:
+- name: \"local\"
+  context:
+    user: \"local\"
+    cluster: \"local\"
+
+current-context: \"local\"
+"
+
+echo "$kubeconfig" >> .temp_kubeconfig.yaml
+# helm will complain if these are group/world readable
+chmod g-r .temp_kubeconfig.yaml
+chmod o-r .temp_kubeconfig.yaml
+export KUBECONFIG="$(pwd)/.temp_kubeconfig.yaml"
+
+if [[ "$tlsVerify" != "" ]]
```

Comment:
This doesn't actually check what the flag is, I could run `adjust-downstream-webhook.sh happy4thofjuly` and it would turn off TLS verification.

I see that the old script worked the same way so maybe it's not in scope to change it here but maybe something to think about later.

Reply:
Good catch!